fix(agw): Check for unset params while extracting status

amarpad opened the pull request:
## Summary

When the eNB just comes up some of the eNB status nodes might not
be populated, check before dereferencing attribute.

Seen in the field right after eNB boots:
Sep 24 02:56:03 FreedomFi-Gateway enodebd[889]:     if name_to_val[cls.ENB_STATUS].upper() == success_str:
Sep 24 02:56:03 FreedomFi-Gateway enodebd[889]: AttributeError: 'NoneType' object has no attribute 'upper'

Signed-off-by: Amar Padmanabhan <amar@freedomfi.com>

<!--
    Tag your PR title with the components that it touches along with
    the type of change
    E.g. "fix(orc8r): Fix reindexer race condition" or "feat(agw) ..."
-->



<!-- Enumerate changes you made and why you made them -->

## Test Plan

Wrote a unit test

## Additional Information

- [ ] This change is backwards-breaking

<!--
    If this is a backwards-breaking change, document the upgrade instructions.
    All upgrade instructions for backwards-breaking changes will be aggregated
    in the next release's changelog so this is very important to fill out.
-->

Commit:
fix(agw): Check for unset params while extracting status

When the eNB just comes up some of the eNB status nodes might not
be populated, check before dereferencing attribute.


Signed-off-by: Amar Padmanabhan <amar@freedomfi.com>

ardzoht commented on lte/gateway/python/magma/enodebd/devices/freedomfi_one.py:
```diff
@@ -233,7 +233,8 @@ def set_magma_device_cfg(
 
-        if name_to_val[cls.DEFAULT_GW].upper() != success_str:
+        if name_to_val[cls.DEFAULT_GW] \
+                and name_to_val[cls.DEFAULT_GW].upper() != success_str:
```
One potential alternative to also do the lookup (and also avoid any `KeyError` exception) it's
```suggestion
        if name_to_val.get(cls.DEFAULT_GW, "").upper() != success_str:
```

However I'm not sure how often could we hit the said exception here (given on how this dictionary is constructed), so I'm fine with the current way of handling this.

amarpad replied:
probably a better suggestion as it helps with keyerror will update

ardzoht reviewed (approved):
lgtm, left minor question

Commit:
fix(agw): Use get insteard of operator []

Address review comments, get is safer than [] and
the code is cleaner as explicitly set to None values are
handled identifical to the key itself being missing.

Signed-off-by: Amar Padmanabhan <amar@freedomfi.com>
